### PR TITLE
deployNodes Gradle task appends properties from an optional file to node.conf

### PR DESCRIPTION
**Pull request by szymonsztuka**
Appending properties from an optional file when deploying node is useful when e.g. testing against different databases.

### CODE REVIEW

File: gradle-plugins/cordformation/src/main/groovy/net/corda/plugins/Node.groovy [hunk 1]
```diff
@@ -184,6 +184,24 @@ class Node extends CordformNode {
         }
         def configFileText = config.root().render(new ConfigRenderOptions(false, false, true, false)).split("\n").toList()
 
+        // Appending properties from an optional file
```
Comment (Clintonio) [inline gradle-plugins/cordformation/src/main/groovy/net/corda/plugins/Node.groovy]: Can this be broken down a bit? Perhaps a separate function on the lines of `getOptionalConfig`/`appendOptionalConfig`? That way this will be a little more maintainable in future. The main goal of this particular function is to actually install the config. 

I am not entirely sure if it's also possible to read the extra config file into memory and then append/merge it to the existing config but if you could we would get a more type safe merge of the files.

File: docs/source/changelog.rst [hunk 2]
```diff
@@ -126,6 +126,9 @@ UNRELEASED
 
 * Test type ``NodeHandle`` now has method ``stop(): CordaFuture<Unit>`` that terminates the referenced node.
 
+* Gradle task ``deployNodes`` can have an additional parameter `configFile` with the path to a properties file
```
Comment (Clintonio) [inline docs/source/changelog.rst]: Can you rephrase this to be "Cordformation node building DSL can...` because deployNodes is just a task with an arbitrary name. It's convention to call it `deployNodes` but it could be anything really.

File: gradle-plugins/cordformation/src/main/groovy/net/corda/plugins/Node.groovy (hunk 1 above)
Comment (szymonsztuka) [reply gradle-plugins/cordformation/src/main/groovy/net/corda/plugins/Node.groovy]: Fixed- code moved from installConfig() to another method

File: constants.properties [hunk 3]
```diff
@@ -1,4 +1,4 @@
-gradlePluginsVersion=0.16.2
+gradlePluginsVersion=0.16.3
```
Comment (Clintonio) [inline constants.properties]: Please change to 1.1.0 or later.

Comment (Clintonio) [reply constants.properties]: Now 2.0.1 or later.

**Comment by Clintonio** [comment]
I'm going to make this PR merge - I have been telling people it exists for a while.
